staging/src/k8s.io/client-go/util/cert/cert.go

```diff
@@ -88,45 +94,55 @@ func NewSelfSignedCACert(cfg Config, key crypto.Signer) (*x509.Certificate, erro
 // GenerateSelfSignedCertKey creates a self-signed certificate and key for the given host.
 // Host may be an IP or a DNS name
 // You may also specify additional subject alt names (either ip or dns names) for the certificate.
-func GenerateSelfSignedCertKey(host string, alternateIPs []net.IP, alternateDNS []string) ([]byte, []byte, error) {
+func GenerateSelfSignedCertKey(host string, alternateIPs []net.IP, alternateDNS []string) (SelfSignedCertKeyWithFixtures, error) {
 	return GenerateSelfSignedCertKeyWithFixtures(host, alternateIPs, alternateDNS, "")
 }
 
-// GenerateSelfSignedCertKeyWithFixtures creates a self-signed certificate and key for the given host.
+// GenerateSelfSignedCertKeyWithFixtures creates a self-signed ca bundle, certificate and key for the given host.
 // Host may be an IP or a DNS name. You may also specify additional subject alt names (either ip or dns names)
 // for the certificate.
 //
 // If fixtureDirectory is non-empty, it is a directory path which can contain pre-generated certs. The format is:
+// <host>_<ip>-<ip>_<alternateDNS>-<alternateDNS>-ca.crt
 // <host>_<ip>-<ip>_<alternateDNS>-<alternateDNS>.crt
 // <host>_<ip>-<ip>_<alternateDNS>-<alternateDNS>.key
 // Certs/keys not existing in that directory are created.
-func GenerateSelfSignedCertKeyWithFixtures(host string, alternateIPs []net.IP, alternateDNS []string, fixtureDirectory string) ([]byte, []byte, error) {
+func GenerateSelfSignedCertKeyWithFixtures(host string, alternateIPs []net.IP, alternateDNS []string, fixtureDirectory string) (SelfSignedCertKeyWithFixtures, error) {
```

Comment:
doesn't the self-signed cert contain the CA already?

Reply:
Yeah this was unnecessary. I am using self-signed cert as it is for the test now.